src/datasets/features/features.py

```diff
@@ -1014,16 +1019,12 @@ def cast_storage(self, storage: Union[pa.StringArray, pa.IntegerArray]) -> pa.In
         """
         if isinstance(storage, pa.IntegerArray):
             min_max = pc.min_max(storage).as_py()
-            if min_max["min"] < -1:
-                raise ValueError(f"Class label {min_max['min']} less than -1")
             if min_max["max"] >= self.num_classes:
                 raise ValueError(
                     f"Class label {min_max['max']} greater than configured num_classes {self.num_classes}"
                 )
         elif isinstance(storage, pa.StringArray):
-            storage = pa.array(
-                [self._strval2int(label) if label is not None else None for label in storage.to_pylist()]
-            )
+            storage = pa.array(self.str2int(storage.to_pylist()))
```

Comment:
Should we support `None` for the string representation of missing labels ? I think it could lead to silent errors in users code if we allow this

Reply:
Yes, I think we should still support this (to be consistent with the behavior during generation - `encode_example` is not called if a label is `None`)

Reply:
Ok ! `None` values can be used for N/A for example (which has a subtle difference compared to "unknown/missing label" represented by negative values).

Though to avoid silent error I would keep the current behavior of str2int and int2str, that is: raise an error if None is given as input

Reply:
I reverted this to keep supporting None here :)